### PR TITLE
[datadog] Add optional kube-state-metrics pod

### PR DESCRIPTION
**Pull request by hkaj**
Optionally deploy kube-state-metrics to get its (awesome!) metrics out of box. It is automatically discovered by the agent running on the same node so no further configuration is required.

This chart is awesome y'all! Thanks for creating it. We will likely list it as the recommended way to install the agent on Kubernetes.

### CODE REVIEW

**Comment by hkaj** [comment]
ping

**Comment by hkaj** [comment]
Can I get a review here, team?

**Comment by elafarge** [comment]
Great addition @hkaj . Can't wait for that to get merged :)

**Comment by roobert** [comment]
@hkaj: I'm somewhat new to Helm so please excuse me if this doesn't make sense.

For pods like kube-state-metrics that are likely dependencies for multiple charts, should they be bundled into individual application charts? The alternative is that we have multiple instances of the same pods running across our infrastructure which seems like a waste of resources.

Cheers,

edit: maybe this would be better off as a separate chart and a _[requirement](https://github.com/kubernetes/helm/blob/master/docs/charts.md#managing-dependencies-with-requirementsyaml)_?

edit 2: looks like there's on open PR for a kube state metrics chart [here](https://github.com/kubernetes/charts/pull/662/files).

**Comment by hkaj** [comment]
@roobert you're right, it's better as another chart. We can switch to a requirement as soon as that PR is merged. And I agree, the risk here is that several charts end up shipping kube-state-metrics (some people use prometheus and datadog, and both will have a kube-state-metrics deployment), that's why I made it optional. This way if you already deploy it another way you can leave it out in the datadog deployment (the datadog agent will still find it and report its metrics),

**Comment by roobert** [comment]
Nice one @hkaj, going to leave a reference here for #662.

**Comment by mgoodness** [comment]
#662 has been merged. There are some differences between the `kube-state-metrics` chart and what's proposed here. I recommend opening a new PR to resolve those differences, then update this PR to use the stand-alone KSM chart as a dependency. We'll get both merged ASAP.

**Comment by hkaj** [comment]
thanks for the feedback @mgoodness 
Updated the PR using requirements (also removed a bunch of changes that became useless without the kube-state-metrics deployment).

**Comment by hkaj** [comment]
@mgoodness Resolved the conflicts, can I get a quick review before new ones pop up?

File: stable/datadog/Chart.yaml [hunk 1]
```diff
@@ -13,3 +13,5 @@ sources:
 maintainers:
 - name: Greg Taylor
   email: gtaylor@gc-taylor.com
+- name: Haissam Kaj
```
Comment (gtaylor) [inline stable/datadog/Chart.yaml]: I don't have the bandwidth to review the rest of this, but wanted to give this the :+1: as the current and only maintainer. The situation should improve greatly in having upstream involved!

**Comment by prydonius** [comment]
@k8s-bot ok to test

**Comment by prydonius** [comment]
thanks a lot @hkaj :)

**Comment by hkaj** [comment]
thanks guys!

**Pull request merged**